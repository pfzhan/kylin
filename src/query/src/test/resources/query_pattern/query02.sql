--
-- Copyright (C) 2020 Kyligence Inc. All rights reserved.
--
-- http://kyligence.io
--
-- This software is the confidential and proprietary information of
-- Kyligence Inc. ("Confidential Information"). You shall not disclose
-- such Confidential Information and shall use it only in accordance
-- with the terms of the license agreement you entered into with
-- Kyligence Inc.
--
-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-- "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-- LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-- A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-- OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-- SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-- LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-- DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-- THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-- OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
--
select l_shipmode,
	sum(case
		when o_orderpriority = '1-URGENT'
			or o_orderpriority = '2-HIGH'
			then 1
		else 0
	end) as high_line_count,
	sum(case
		when o_orderpriority <> '1-URGENT'
			and o_orderpriority <> '2-HIGH'
			then 1
		else 0
	end) as low_line_count
from tpch.lineitem inner join tpch.orders on o_orderkey = l_orderkey
where
	l_shipmode in ('REG AIR', 'MAIL')
	and l_commitdate < l_receiptdate
	and l_shipdate < l_commitdate
	and l_receiptdate >= '1995-01-01'
	and l_receiptdate < '1996-01-01'
group by l_shipmode
order by l_shipmode