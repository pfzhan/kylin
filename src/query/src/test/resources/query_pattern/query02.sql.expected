SELECT "L_SHIPMODE", SUM((CASE WHEN (("O_ORDERPRIORITY" = '1-URGENT') OR ("O_ORDERPRIORITY" = '2-HIGH')) THEN 1 ELSE 0 END)) "HIGH_LINE_COUNT", SUM((CASE WHEN (("O_ORDERPRIORITY" <> '1-URGENT') AND ("O_ORDERPRIORITY" <> '2-HIGH')) THEN 1 ELSE 0 END)) "LOW_LINE_COUNT"
FROM "TPCH"."LINEITEM"
INNER JOIN "TPCH"."ORDERS" ON ("O_ORDERKEY" = "L_ORDERKEY")
WHERE ((((("L_SHIPMODE" IN ('A')) AND ("L_COMMITDATE" < "L_RECEIPTDATE")) AND ("L_SHIPDATE" < "L_COMMITDATE")) AND ("L_RECEIPTDATE" >= '2010-01-01')) AND ("L_RECEIPTDATE" < '2010-01-02'))
GROUP BY "L_SHIPMODE"
ORDER BY "L_SHIPMODE"